[#342] Add guide on how to connect Rasa and fix webhook API (#423)
airyhq/airy: docs/docs/guides/airy-core-and-rasa.md

- A local Rasa setup: For convenience, we recommend the Docker one [[Guide](https://rasa.com/docs/rasa/docker/building-in-docker/ )] or [this demo repository](https://github.com/airyhq/rasa-demo) we created for this guide

## Configuring Airy

In this scenario the Airy Core Platform works like a message router: It receives
messages from your sources (Messenger, WhatsApp etc.) and forwards them to downstream messaging frameworks like Rasa, which can in turn reply using Airy.

To do this we follow the [Webhook documentation](api/webhook.md) to forward in- and
outbound messages to Rasa. When doing so set the `url` parameter
so that it points to your Rasa installation. If it is running locally you can obtain
a public url by using [ngrok](https://ngrok.com/). Run: 

```shell script
ngrok http 5005
```

to get an ngrok URL that points to your local Rasa installation at `localhost:5005`.
With the public ngrok url your Webhook subscription payload should look like so:

```json
{
  "url": "https://e3effaceb9c5.ngrok.io/webhooks/airy/webhook"
}
``` 

Once you have done this, the Core Platform will start sending all messages to the
url you specified.

## Configuring Rasa

> Rasa comes with a default rest connector to consume HTTP webhooks. But since we 
> also want to use the Airy Api to respond to messages we will be implementing a
> [custom connector](https://rasa.com/docs/rasa/connectors/custom-connectors/).

Reviewer: ```suggestion
We will be implementing a
[custom connector](https://rasa.com/docs/rasa/connectors/custom-connectors/) as we want to make use of the Airy Core Platform API.
```

> For this we create a new directory `channels/` in our Rasa project and copy this

Reviewer: ```suggestion
The first step is to create a new directory `channels/` in our Rasa project and copy this
```

> [connector file](https://github.com/airyhq/rasa-demo/blob/master/channels/airy.py) into it. This Airy connector requires two configuration values to be set so that it can communicate
> with the Core Platform API: 

Reviewer: ```suggestion
[connector file](https://github.com/airyhq/rasa-demo/blob/master/channels/airy.py) into it. The connector requires the following configuration values: 
```

> - `auth_token`  This is the Airy Core Platform jwt token, which you used

Reviewer: ```suggestion
- `auth_token`  the Airy Core Platform JWT token you used
```

                to connect the webhook.

> - `api_host`    The url, where you host your Airy Core Platform API. For a local installation
>                 this would be `http://api.airy`.

Reviewer: ```suggestion
- `api_host`    The url where you host your Airy Core Platform API (`http://api.airy` for a local installation).
```

> Add these to your existing Rasa `credentials.yml` like so:

Reviewer: ```suggestion
Add them to your existing Rasa `credentials.yml`:
```

```yaml
channels.airy.AiryInput:

>   api_host: "your jwt authentication token"

Reviewer: ```suggestion
  api_host: "your JWT authentication token"
```

  auth_token: "http://api.airy"
``` 

> With this you are already done. To test the connection write a message to one of your sources
> that is recognized by your Rasa model.

Reviewer: ```suggestion
Now you should have a working integration 🎉 . To test the connection write a message to one of the sources
recognized by your Rasa model.
```
